packages/typespec-ts/src/utils/clientUtils.ts

```diff
@@ -0,0 +1,16 @@
+import {
+  SdkClient,
+  SdkContext,
+  listClients
+} from "@azure-tools/typespec-client-generator-core";
+
+export function getRLCClients(dpgContext: SdkContext): SdkClient[] {
+  const clientsInSdk = listClients(dpgContext);
+  const clients = listClients(dpgContext, false);
+  // to respect the boundaries that is defined by one @client
+  if (clientsInSdk.length === 1) {
+    return clientsInSdk;
+  }
+  // return the original @service clients
+  return clients;
+}
```

Comment:
This is the part how we get the RLCClient now.

Reply:
If the clientsInSdk.length === 1, it means, there's only one @client decorator, and we should respect the bondaries defined by it.  This is working because we already has test cases like healthinsights https://github.com/Azure/autorest.typescript/tree/main/packages/typespec-test/test/healthinsight and there are no changes from healthinsight in this PR which is expected.